src/main/java/org/craftercms/studio/impl/v2/service/content/ContentServiceImpl.java

```diff
@@ -230,79 +232,76 @@ public DetailedItem getItemById(@ProtectedResourceId(SITE_ID_RESOURCE_ID) String
 
     @Override
     @HasPermission(type = CompositePermission.class, action = PERMISSION_CONTENT_WRITE)
-    public void itemLockByPath(@ProtectedResourceId(SITE_ID_RESOURCE_ID) String siteId,
-                               @ProtectedResourceId(PATH_RESOURCE_ID) String path)
+    public void contentLockByPath(@ProtectedResourceId(SITE_ID_RESOURCE_ID) String siteId,
+                                  @ProtectedResourceId(PATH_RESOURCE_ID) String path)
             throws UserNotFoundException, ServiceLayerException {
-        var item = itemServiceInternal.getItem(siteId, path);
-        if (Objects.nonNull(item)) {
-            var username = securityService.getCurrentUser();
-            if (StringUtils.isEmpty(item.getLockOwner())) {
-                contentServiceInternal.itemLockByPath(siteId, path);
-                itemServiceInternal.lockItemByPath(siteId, path, username);
-            } else {
-                if (!StringUtils.equals(item.getLockOwner(), username)) {
-                    var e = new ContentLockedByAnotherUserException();
-                    e.setLockOwner(username);
-                    throw e;
+        String lockKey = siteId + ":" + path;
```

Comment:
Add method to GeneralLockService to generate this type of "item key"
GenerateItemKey(String site, String path)